src/features/dashboard/Dashboard.tsx

```diff
@@ -1,15 +1,44 @@
-import { Link } from 'react-router-dom'
-import Box  from '@mui/material/Box'
-import Typography  from '@mui/material/Typography'
+import { Link, useNavigate } from "react-router-dom";
+import Typography from "@mui/material/Typography";
+import Button from "@mui/material/Button";
+import { Box } from "@mui/material";
+import { useDispatch, useSelector } from "react-redux";
+import { AuthState } from "features/auth/store/types";
+import { User } from "core/users/types";
+import useRedirectWhenUnlogged from "hooks/use-redirect-when-unlogged";
+import { logout } from "features/auth/store/auth-slice";
+import { buttonStyles, dashboardBoxStyles } from "./styles/styles";
 
 const Dashboard = () => {
+  const user = useCurrentUser();
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+
+  useRedirectWhenUnlogged();
+
+  if (!user) {
+    return null;
+  }
+
+  const onLogout = () => {
+    dispatch(logout());
+    navigate("/home");
+  };
```

Comment:
Te recomiendo que siempre que pases una funcion como prop sea un `useCallback`
```suggestion
  const onLogout = useCallback(() => {
    dispatch(logout());
    navigate("/home");
  }, []);
```